app/models/incomeSourceDetails/TaxYear.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.incomeSourceDetails
+
+case class TaxYear(startYear: String, endYear: String)
```

Comment:
assuming you are doing all required validation and converting string into your custom type, I think this would make more sense to keep startYear and endYear as Integer + you might need to add:
required(startYear + 1 == endYear), as this is more accurate description of the data for this new type